workspaces/blackduck/plugins/blackduck-backend/src/service/BlackDuckConfig.ts

```diff
@@ -0,0 +1,48 @@
+import { Config } from '@backstage/config';
+
+/**
+ * @public
+ */
+export interface BlackDuckHostConfig {
+  name: string;
+  host: string;
+  token: string;
+}
+
+/**
+ * blackduckPlugin config
+ *
+ * @public
+ */
+export class BlackDuckConfig {
+  constructor(private readonly hosts: BlackDuckHostConfig[]) {}
+
+  static fromConfig(config: Config): BlackDuckConfig {
+    const defaultHost = {
+      name: 'default',
+      host: config.getString('blackduck.host'),
+      token: config.getString('blackduck.token'),
+    };
+
+    const hosts = [
+      defaultHost,
+      ...(config.getOptionalConfigArray('blackduck.hosts')?.map(host => ({
+        name: host.getString('name'),
+        host: host.getString('host'),
+        token: host.getString('token'),
+      })) ?? []),
+    ];
+
+    return new BlackDuckConfig(hosts);
+  }
+
+  getHostConfigByName(name: string): BlackDuckHostConfig {
+    const hostConfig = this.hosts.find(host => host.name === name);
```

Comment:
Check if the name is `default` (for non host specified annotation) go for the config `default` name, else go for the named host instance